File: v1/src/main/java/com/api/v1/persons/dtos/PersonModificationRequestDto.java
package com.api.v1.persons.dtos;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

import java.time.LocalDate;

public record PersonModificationRequestDto(
        @NotBlank String firstName,
        String middleName,
        @NotBlank String lastName,
        LocalDate birthDate,
        @NotBlank @Email String email,
        @NotBlank String address,
        @NotBlank @Size(min = 10, max = 10) String phoneNumber,
        @NotBlank @Size(min = 1) String gender
) {
}

File: kotlin/v3/src/main/kotlin/com/api/v3/persons/dtos/PersonModificationRequestDto.kt
package com.api.v3.persons.dtos

import java.time.LocalDate

data class PersonModificationRequestDto(
    val firstName: String,
    val middleName: String?,
    val lastName: String,
    val email: String,
    val birthDate: LocalDate,
    val address: String,
    val gender: String,
    val phoneNumber: String
)
